### ISSUE TITLE
HomeyAPI.ManagerInsights.getEntries gives timeout for large datasets (and Homey freezes)

### ISSUE THREAD

**Issue by gruijter** [athombv/homey-issues-dp#135]
**What is your Homey version (Settings → System)?**

1.5.6-rc.4

**What did you try to achieve?**

use the web api to retrieve log entries. This works ok for small logs, but does not work for logs with a very large set of entries (6 months of data @10s interval, approx. 1.5 million entries in one log)

*Please include detailed steps how we can reproduce the issue*

```
const { HomeyAPI } = require('athom-api');
...
async onInit() {
this.api = await HomeyAPI.forCurrentHomey();
...
const entries = await this.api.insights.getEntries(opts);
```

**What was the result?**

```
{ Error: Timed out after 30000ms.
    at Timeout.<anonymous> (/node_modules/athom-api/dist/index.js:1:865791)
    at ontimeout (timers.js:475:11)
    at Timer.unrefdHandle (timers.js:588:5) code: 'TIMEOUT' }
```

And secondly, Homey freezes for more then 10 minutes. This means it is offline / not reachable.

**What did you expect as result?**

* To avoid a timeout error, and/or to be able to get the entries as a stream.
* Homey not to freeze.

**Any other remarks**
Homey also freezes for more then 10 minutes when I want to display the Insights for a long period (6 months). In the end it does display the entire 6 months though. When trying to download the data from the insights tab It also freezes, then starts downloading several MB, and eventually fails to finish the download. 
...

**Comment by jeroenvollenbrock** [comment]
The newest version of the API client supports a start&end option so you can retrieve a timestamp-based selection of entries. To reduce the freezing, i'll investigate if it's possible to optimize the call in a future version of Homey as well.

**Comment by jeroenvollenbrock** [comment]
@gruijter Updated docs can be found here: https://developer.athom.com/docs/api/HomeyAPI.ManagerInsights.html#getEntries

**Comment by gruijter** [comment]
Allas, also with athom-api v 2.0.92 I get a timeout, even if selecting only a single day. This only happens for logs that have a lot of data (e.g. system log), but not for smaller logs (e.g. nl.philips.hue log)

**Comment by jeroenvollenbrock** [comment]
rc9/rc10 contains a possible fix for this, can you test again?

**Comment by gruijter** [comment]
Using athom-api 2.0.96 and Homey 1.5.6-rc.10 the timeout problem is gone 👍 

BUT: during the update to rc10 all insights before 29.1.2018 got deleted somehow. So my tests are based on only 4 days of log entries. I will have to test in a couple of weeks again to see if I still do not get any timeouts.

**Comment by gruijter** [comment]
😯
Now after a few more days of logs the timeout happens again...
```
Error: Timed out after 30000ms. at Timeout. (/node_modules/athom-api/dist/index.js:1:890859) at ontimeout (timers.js:475:11) at Timer.unrefdHandle (timers.js:588:5) code: 'TIMEOUT'
```

**Comment by jeroenvollenbrock** [comment]
How about rc12/13?

**Comment by gruijter** [comment]
on rc 10 they happened sporadically. On rc13 they increase a lot.
```
{ n: response timeout at http://localhost:80/api/manager/insights/log/homey:device:8bf85b9c-77e5-4f62-b9bd-35b11908894a/measure_power/entry over limit: 30000
    at Timeout._onTimeout (/node_modules/athom-api/dist/index.js:1:191548)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
    at Timer.listOnTimeout (timers.js:270:5)
  name: 'n',
  message: 'response timeout at http://localhost:80/api/manager/insights/log/homey:device:8bf85b9c-77e5-4f62-b9bd-35b11908894a/measure_power/entry over limit: 30000',
  type: 'body-timeout' }
```

**Comment by gruijter** [comment]
adding $timeout:120000 to the opts object fixed the issue for now. This only wotks for local connection I believe. So for cloud connection there is still no real solution for large datasets (larger then 64000 entries)

**Comment by gruijter** [comment]
No, this is definitely still not fixed (I am now on 1.5.9) A timeout of 120000 is not enough. Furthermore Homey freezes, and it causes several apps to crash due to IPC_CHANNEL_CLOSED errors.
